src/main/java/school/lemon/changerequest/java/io/hw1/IFileManagerImp.java

```diff
@@ -0,0 +1,625 @@
+                case BY_LEXICOGRAPHICAL_ASCENDING:
+                    folder = new File(getPath());
+                    listOfFiles = folder.listFiles(fileFilter);
+                    Arrays.sort(listOfFiles, new Comparator<File>() {
+                        @Override
+                        public int compare(File o1, File o2) {
+                            return String.CASE_INSENSITIVE_ORDER.compare(o1.getName(), o2.getName());
+                        }
+                    });
+                    for (File file : listOfFiles) {
+                        sb.append(file.getName());
+                        sb.append(" ");
+                    }
+                    return sb.toString();
+                case BY_LEXICOGRAPHICAL_DESCENDING:
+                    folder = new File(getPath());
+                    listOfFiles = folder.listFiles(fileFilter);
+                    Arrays.sort(listOfFiles, new Comparator<File>() {
+                        @Override
+                        public int compare(File o1, File o2) {
+                            return String.CASE_INSENSITIVE_ORDER.compare(o2.getName(), o1.getName());
+                        }
+                    });
+                    for (File file : listOfFiles) {
+                        sb.append(file.getName());
+                        sb.append(" ");
+                    }
+                    return sb.toString();
+
+
+                case BY_LAST_UPDATE_DESCENDING:
+                    folder = new File(getPath());
+                    listOfFiles = folder.listFiles(fileFilter);
+                    Arrays.sort(listOfFiles, new Comparator<File>() {
+                        @Override
+                        public int compare(File o1, File o2) {
+                            return Long.valueOf(o1.lastModified()).compareTo(o2.lastModified());
+                        }
+                    });
+                    for (File file : listOfFiles) {
+                        sb.append(file.getName());
+                        sb.append(" ");
+                    }
+                    return sb.toString();
+                case BY_LAST_UPDATE_ASCENDING:
+                    folder = new File(getPath());
+                    listOfFiles = folder.listFiles(fileFilter);
+                    Arrays.sort(listOfFiles, new Comparator<File>() {
+                        @Override
+                        public int compare(File o1, File o2) {
+                            return Long.valueOf(o2.lastModified()).compareTo(o1.lastModified());
+                        }
+                    });
+                    for (File file : listOfFiles) {
+                        sb.append(file.getName());
+                        sb.append(" ");
+                    }
+                    return sb.toString();
+            }
+        }
+        return null;
+    }
+
+
+    @Override
+    public List<File> filter(Order order, boolean directoriesFirst, FileFilter fileFilter) {
```

Comment:
You've got too much duplicated code - please think how could you avoid duplication of your comparison logic.
There is actually no need to create new Comparator object each time, you can define some constant with pre-defined comparators and reuse them.